src/helpers/joi.adapter.js

```diff
@@ -0,0 +1,46 @@
+const Joi = require('joi');
+const _ = require('lodash');
+
+const joiOptions = {
+  abortEarly: false,
+  allowUnknown: true,
+  stripUnknown: {
+    objects: true,
+  },
+};
+
+/**
+ * Parse and return list of errors
+ * @param {object} joiError
+ * @return {object[]}
+ */
+const parseJoiErrors = (joiError) => {
+  let resultErrors = [];
+  if (joiError && _.isArray(joiError.details)) {
+    resultErrors = joiError.details.map((error) => {
+      const pathLastPart = error.path.slice(error.path.length - error.context.key.length);
+
+      if (pathLastPart === error.context.key) {
+        return { [error.path]: error.message };
+      }
+
+      return { [error.context.key]: error.message };
+    });
+  }
+
+  return resultErrors;
+};
+
+const validate = _.curry((schema, payload) => {
+  const { error, value } = Joi.validate(payload, schema, joiOptions);
+
+  return {
+    errors: parseJoiErrors(error),
+    value,
+  };
+});
+
+module.exports = {
+  ...Joi,
+  validate,
+};
```

Comment:
I'm not a specialist in GoF patterns. But for me it looks like `decorator`, not `adapter`. But I'm not sure here.

Reply:
It's a decorator, it is an adapter. It's *Batman* :).  I'm not a specialist too. It is really not an adapter. Maybe it is a proxy.